java/src/jmri/util/FileUtilSupport.java

```diff
@@ -485,19 +510,25 @@ public boolean isPortableFilename(String filename) {
      *
      * @return User's home directory as a String
      */
+    @Nonnull
+    @CheckReturnValue
     public String getHomePath() {
         return HOME_PATH;
     }
 
     /**
      * Get the user's files directory. If not set by the user, this is the same
-     * as the profile path.
+     * as the profile path; if all else fails, the program directory. 
      *
      * @see #getProfilePath()
-     * @return User's files directory as a String
+     * @return User's files directory as a String - never null
      */
+    @Nonnull
+    @CheckReturnValue
     public String getUserFilesPath() {
-        return (this.userFilesPath != null) ? this.userFilesPath : this.getProfilePath();
+        if (userFilesPath != null) return userFilesPath;
+        if (getProfilePath() != null) return getProfilePath();
+        return getProgramPath();
```

Comment:
This should not return the program path as a fall back location -- we do not ever want JMRI writing into its own installation directory. It would be better to fall back to the settings location.